src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordEquals.cs

```diff
@@ -125,14 +125,27 @@ internal override void GenerateMethodBody(TypeCompilationState compilationState,
 
                 // field1 == other.field1 && ... && fieldN == other.fieldN
                 var fields = ArrayBuilder<FieldSymbol>.GetInstance();
+                bool foundBadField = false;
                 foreach (var f in ContainingType.GetFieldsToEmit())
                 {
                     if (!f.IsStatic)
                     {
                         fields.Add(f);
+
+                        var parameterType = f.Type;
+                        if (parameterType.IsUnsafe())
+                        {
+                            diagnostics.Add(ErrorCode.ERR_BadFieldTypeInRecord, f.Locations[0], parameterType);
```

Comment:
>[0] [](start = 91, length = 3)

Since we do not control what kind of symbols we are going to get here, consider using FirstOrNone helper here.